minio/datatypes.py

```diff
@@ -775,8 +775,12 @@ def form_data(self, creds, region):
         policy["conditions"].append([_EQ, "$x-amz-date", amz_date])
 
         policy = base64.b64encode(json.dumps(policy).encode())
+        if isinstance(policy, (bytes, bytearray)):
+            policy = policy.decode()  # convert policy to string
+        if isinstance(policy, str) is not True:
```

Comment:
As `base64.b64encode()` always return `bytes`, you would need to directly decode to string. It is better to fail than type checking in python.

Reply:
ok, it make sense, thank you Bala